crypto/crypto.go

```diff
@@ -79,6 +79,20 @@ func txIDFromTransaction(tx types.Transaction) (txid string) {
 	return
 }
 
+// TransactionID is the unique identifier for a Transaction in progress
```

Comment:
Does this duplicate `txIDFromTransaction` above? / could the usages of `txIDFromTransaction` be reworked to use this new public helper?